protopipe/scripts/write_dl2.py

```diff
@@ -206,181 +251,285 @@ class RecoEvent(tb.IsDescription):
-                reco_result.az,
-                reco_result.alt,
-            )
+                # Angular separation between
+                # - true direction
+                # - reconstruted direction
+                xi = angular_separation(
+                    event.mc.az, event.mc.alt, reco_result.az, reco_result.alt
+                )
 
-            # Height of shower maximum
-            h_max = reco_result.h_max
-
-            if hillas_dict is not None:
-
-                # Estimate particle energy
-                if use_regressor is True:
-                    energy_tel = np.zeros(len(hillas_dict.keys()))
-                    weight_tel = np.zeros(len(hillas_dict.keys()))
-
-                    for idx, tel_id in enumerate(hillas_dict.keys()):
-                        cam_id = event.inst.subarray.tel[tel_id].camera.cam_id
-                        moments = hillas_dict[tel_id]
-                        model = regressor.model_dict[cam_id]
-
-                        # Features to be fed in the regressor
-                        features_img = np.array(
-                            [
-                                np.log10(moments.intensity),
-                                np.log10(impact_dict[tel_id].value),
-                                moments.width.value,
-                                moments.length.value,
-                                h_max.value,
-                            ]
-                        )
+                # Angular separation between
+                # - center of the array's FoV
+                # - reconstructed direction
+                offset = angular_separation(
+                    run_array_direction[0],  # az
+                    run_array_direction[1],  # alt
+                    reco_result.az,
+                    reco_result.alt,
+                )
+
+                # Reconstructed height of shower maximum
+                h_max = reco_result.h_max
+
+                # Reconstructed position of the shower's core on the ground
+                reco_core_x = reco_result.core_x
+                reco_core_y = reco_result.core_y
+
+                # Reconstructed direction of the shower's in the sky
+                alt, az = reco_result.alt, reco_result.az
 
+                # Successfully reconstructed shower
+                is_valid = True
+
+            else:  # no successful reconstruction assign dummy values
+
+                xi = np.nan * u.deg
+                offset = np.nan * u.deg
+                reco_core_x = np.nan * u.m
+                reco_core_y = np.nan * u.m
+                h_max = np.nan * u.m
+                alt = np.nan * u.deg
+                az = np.nan * u.deg
+                is_valid = False
+                reco_energy = np.nan
+                score = np.nan
+                gammaness = np.nan
+                reco_event["success"] = False
+
+            # Estimate particle energy
+            if (use_regressor is True) and is_valid:
```

Comment:
can just say `if use_regressor and is_valid:`

Reply:
done thanks